azure.go

```diff
@@ -52,7 +55,6 @@ type azureProvider struct {
 	oidcVerifier         *oidc.IDTokenVerifier
 	settings             *azureSettings
 	httpClient           *http.Client
-	authorizer           autorest.Authorizer
 	authorizerExpiration time.Time
 	lock                 sync.RWMutex
```

Comment:
Looks unused now? We might want to understand why this was here in the first place to see if we can remove it.

Reply:
This was previously in the `getAuthorizer` file where we were doing the following access logic: https://github.com/hashicorp/vault-plugin-auth-azure/blob/main/azure.go#L162-L178

I think you're right and we can safely remove this. Thanks for the callout!